api/client/pull.go

```diff
@@ -1,74 +1,14 @@
 package client
 
 import (
-	"errors"
-	"fmt"
-
 	"golang.org/x/net/context"
 
-	Cli "github.com/docker/docker/cli"
 	"github.com/docker/docker/pkg/jsonmessage"
-	flag "github.com/docker/docker/pkg/mflag"
-	"github.com/docker/docker/reference"
-	"github.com/docker/docker/registry"
 	"github.com/docker/engine-api/types"
 )
 
-// CmdPull pulls an image or a repository from the registry.
-//
-// Usage: docker pull [OPTIONS] IMAGENAME[:TAG|@DIGEST]
-func (cli *DockerCli) CmdPull(args ...string) error {
-	cmd := Cli.Subcmd("pull", []string{"NAME[:TAG|@DIGEST]"}, Cli.DockerCommands["pull"].Description, true)
-	allTags := cmd.Bool([]string{"a", "-all-tags"}, false, "Download all tagged images in the repository")
-	addTrustedFlags(cmd, true)
-	cmd.Require(flag.Exact, 1)
-
-	cmd.ParseFlags(args, true)
-	remote := cmd.Arg(0)
-
-	distributionRef, err := reference.ParseNamed(remote)
-	if err != nil {
-		return err
-	}
-	if *allTags && !reference.IsNameOnly(distributionRef) {
-		return errors.New("tag can't be used with --all-tags/-a")
-	}
-
-	if !*allTags && reference.IsNameOnly(distributionRef) {
-		distributionRef = reference.WithDefaultTag(distributionRef)
-		fmt.Fprintf(cli.out, "Using default tag: %s\n", reference.DefaultTag)
-	}
-
-	var tag string
-	switch x := distributionRef.(type) {
-	case reference.Canonical:
-		tag = x.Digest().String()
-	case reference.NamedTagged:
-		tag = x.Tag()
-	}
-
-	registryRef := registry.ParseReference(tag)
-
-	// Resolve the Repository name from fqn to RepositoryInfo
-	repoInfo, err := registry.ParseRepositoryInfo(distributionRef)
-	if err != nil {
-		return err
-	}
-
-	ctx := context.Background()
-
-	authConfig := cli.ResolveAuthConfig(ctx, repoInfo.Index)
-	requestPrivilege := cli.RegistryAuthenticationPrivilegedFunc(repoInfo.Index, "pull")
-
-	if isTrusted() && !registryRef.HasDigest() {
-		// Check if tag is digest
-		return cli.trustedPull(ctx, repoInfo, registryRef, authConfig, requestPrivilege)
-	}
-
-	return cli.imagePullPrivileged(ctx, authConfig, distributionRef.String(), requestPrivilege, *allTags)
-}
-
-func (cli *DockerCli) imagePullPrivileged(ctx context.Context, authConfig types.AuthConfig, ref string, requestPrivilege types.RequestPrivilegeFunc, all bool) error {
+// ImagePullPrivileged pulls the image and displays it to the output
+func (cli *DockerCli) ImagePullPrivileged(ctx context.Context, authConfig types.AuthConfig, ref string, requestPrivilege types.RequestPrivilegeFunc, all bool) error {
```

Comment:
@dnephin yeah I was wondering about that, same goes for `push.go`. I think I'll move it to `api/client/trust.go` for now 😇